spec/features/cart/checkout_spec.rb

```diff
@@ -1,39 +1,199 @@
+      @pencil = @mike.items.create!(name: "Yellow Pencil", description: "You can write on paper with it!", price: 2, image: "https://images-na.ssl-images-amazon.com/images/I/31BlVr01izL._SX425_.jpg", inventory: 100)
+
+      2.times do
+        visit "/items/#{@paper.id}"
+        click_on "Add To Cart"
+      end
 
-      @tire = @meg.items.create(name: "Gatorskins", description: "They'll never pop!", price: 100, image: "https://www.rei.com/media/4e1f5b05-27ef-4267-bb9a-14e35935f218?size=784x588", inventory: 12)
-      @paper = @mike.items.create(name: "Lined Paper", description: "Great for writing on!", price: 20, image: "https://cdn.vertex42.com/WordTemplates/images/printable-lined-paper-wide-ruled.png", inventory: 3)
-      @pencil = @mike.items.create(name: "Yellow Pencil", description: "You can write on paper with it!", price: 2, image: "https://images-na.ssl-images-amazon.com/images/I/31BlVr01izL._SX425_.jpg", inventory: 100)
-      visit "/items/#{@paper.id}"
-      click_on "Add To Cart"
       visit "/items/#{@tire.id}"
       click_on "Add To Cart"
-      visit "/items/#{@pencil.id}"
-      click_on "Add To Cart"
+
+      3.times do
+        visit "/items/#{@pencil.id}"
+        click_on "Add To Cart"
+      end
+
       @items_in_cart = [@paper,@tire,@pencil]
     end
 
-    it 'Theres a link to checkout' do
+    it "requires a visitor to login or register before checking out" do
       visit "/cart"
 
-      expect(page).to have_link("Checkout")
+      expect(page).to_not have_link("Checkout")
+      expect(page).to have_content("You must Register or Log In to finish checking out.")
+      expect(page).to have_link("Register")
+      expect(page).to have_link("Log In")
+    end
 
-      click_on "Checkout"
+    it "has a Register link which links to registration page" do
+      visit "/cart"
 
-      expect(current_path).to eq("/orders/new")
+      within '#checkout-authorization' do
+        click_on "Register"
+      end
+
+      expect(current_path).to eq("/register")
     end
-  end
 
-  describe 'When I havent added items to my cart' do
-    it 'There is not a link to checkout' do
+    it "has a Log In link which links to registration page" do
       visit "/cart"
 
-      expect(page).to_not have_link("Checkout")
+      within '#checkout-authorization' do
+        click_on "Log In"
+      end
+
+      expect(current_path).to eq("/login")
+    end
+
+    describe 'when logged in as a regular user' do
+      before do
+        @user = User.create!(name:"Jackie Chan", address:"skdjfhdskjfh", city:"kajshd", state:"jsdh", zip:"88888", email: "12345@gmail.com", password:"Iamapassword", password_confirmation:"Iamapassword", role: 0)
+
+        visit "/"
+
+        within 'nav' do
+          click_link "Login"
+        end
+
+        fill_in :email, with: @user.email
+        fill_in  :password, with: @user.password
+        click_button "Login"
+
+        visit "/cart"
```

Comment:
Let's try to implement this to get rid of lines 71-79:
`allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@bike_shop_employee)`

Reply:
Updated by @LHJE